src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -179,10 +185,22 @@ IOptions<FeatureConfiguration> featureConfiguration
 
             if (_enableDropInvalidDicomJsonMetadata)
             {
-                // drop invalid metadata
+
                 foreach (DicomTag tag in storeValidatorResult.InvalidTagErrors.Keys)
                 {
+                    // drop invalid metadata
                     dicomDataset.Remove(tag);
+
+                    InstanceIdentifier instanceIdentifier = new InstanceIdentifier(
```

Comment:
If you have the dataset, you can do````suggestion
                    dicomDataset.ToInstanceIdentifier()
```